Sources/SwiftFormat/PrettyPrint/PrettyPrint.swift

```diff
@@ -479,6 +479,9 @@ public class PrettyPrinter {
         }
       }
       outputBuffer.write(comment.print(indent: currentIndentation))
+      if wasEndOfLine, (comment.kind == .line || comment.kind == .docLine) {
```

Comment:
I suppose this _works_, but I'm more worried about the token stream creation itself (I should have been more specific in my earlier comment). Usually the newline behavior is controlled by what kind of `break` is inserted into the token stream, and I would expect that a line comment would always be preceded by something whose newline behavior is `soft` (meaning always insert a newline, unless it would contribute more than the maximum number of blank lines allowed). The original issue would seem to imply that that's not the case, and I think we need to understand why.

Reply:
Thanks for the feedback!
The immediate cause of the ignored `soft` newline was the logic that suppresses line breaks via `printerControl` tokens when visiting an `ImportDeclSyntax`.

As a result, the following code triggers a formatting issue:
```swift
@foo // comment
@bar
import Baz
```
Whereas this code (with a non-import declaration) does not:

```swift
@foo // comment
@bar
let var = 0
```
Looking at #501, this suppression behavior appears to be intentional.
So instead of changing the suppression logic itself, I modified it to ignore suppression when the token immediately preceding the break is a `//` or `///` comment.